leetcodePython/BinarySearch/bs_1146.py

```diff
@@ -0,0 +1,45 @@
+import bisect
+
+
+class SnapshotArray_BruteForce:
+    def __init__(self, length: int):
+        self._snap_id = -1
+        self._array = [0] * length
+        self.snap_info = {}
+
+    def set(self, index: int, val: int) -> None:
+        self._array[index] = val
+
+    def snap(self) -> int:
+        self._snap_id += 1
+        self.snap_info[self._snap_id] = [val for val in self._array]
```

Comment:
Every time you snap you copy all data, but they might be all same. This is the point we can do optimization. We only copy data when we `set data`